envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/routing/ServiceTagPreferenceIngressTest.kt

```diff
@@ -0,0 +1,69 @@
+package pl.allegro.tech.servicemesh.envoycontrol.routing
+
+import okhttp3.Headers.Companion.headersOf
+import okhttp3.Headers.Companion.toHeaders
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.consul.ConsulExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoycontrol.EnvoyControlExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.service.HttpsEchoExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.service.asHttpsEchoResponse
+
+class ServiceTagPreferenceIngressTest {
+
+    companion object {
+        @RegisterExtension
+        val consul = ConsulExtension()
+
+        @RegisterExtension
+        val envoyControl = EnvoyControlExtension(
+            consul = consul, mapOf(
+                "envoy-control.envoy.snapshot.routing.service-tags.enabled" to true,
+                "envoy-control.envoy.snapshot.routing.service-tags.preference-routing.header" to "x-service-tag-preference",
+                "envoy-control.envoy.snapshot.routing.service-tags.preference-routing.default-preference-env" to "DEFAULT_SERVICE_TAG_PREFERENCE",
+                "envoy-control.envoy.snapshot.routing.service-tags.preference-routing.default-preference-fallback" to "global",
+                "envoy-control.envoy.snapshot.routing.service-tags.preference-routing.enable-for-all" to true,
+            )
+        )
+
+        @RegisterExtension
```

Comment:
Looks like `@JvmField` annotation is no longer needed. I don't know why, but I'm OK with it.